test/built-ins/Array/prototype/toLocaleString/resizable-buffer.js

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.prototype.tolocalestring
+description: >
+  Array.p.toLocaleString behaves correctly on TypedArrays backed by resizable
+  buffers.
+includes: [resizableArrayBufferUtils.js]
+features: [resizable-arraybuffer]
+---*/
+
+for (let ctor of ctors) {
+  const rab = CreateResizableArrayBuffer(4 * ctor.BYTES_PER_ELEMENT, 8 * ctor.BYTES_PER_ELEMENT);
+  const fixedLength = new ctor(rab, 0, 4);
+  const fixedLengthWithOffset = new ctor(rab, 2 * ctor.BYTES_PER_ELEMENT, 2);
+  const lengthTracking = new ctor(rab, 0);
+  const lengthTrackingWithOffset = new ctor(rab, 2 * ctor.BYTES_PER_ELEMENT);
+  const taWrite = new ctor(rab);
+
+  // Write some data into the array.
+  for (let i = 0; i < 4; ++i) {
+    WriteToTypedArray(taWrite, i, 2 * i);
+  }
+
+  // Orig. array: [0, 2, 4, 6]
+  //              [0, 2, 4, 6] << fixedLength
+  //                    [4, 6] << fixedLengthWithOffset
+  //              [0, 2, 4, 6, ...] << lengthTracking
+  //                    [4, 6, ...] << lengthTrackingWithOffset
+
+  assert.sameValue(Array.prototype.toLocaleString.call(fixedLength), '0,2,4,6');
+  assert.sameValue(Array.prototype.toLocaleString.call(fixedLengthWithOffset), '4,6');
+  assert.sameValue(Array.prototype.toLocaleString.call(lengthTracking), '0,2,4,6');
+  assert.sameValue(Array.prototype.toLocaleString.call(lengthTrackingWithOffset), '4,6');
```

Comment:
As per https://tc39.es/ecma262/#sec-array.prototype.tolocalestring, we are assuming the separator is `","` even though it's implementation-defined.

In fact, the spec text itself suggests using `", "` with a space. Engine262 does this, while all other engines that I tested use `","` without a space.

I think we might have to write these tests to be a bit more agnostic to the localized output.

Note you can get the separator by using `[undefined,undefined].toLocaleString()`, and we should also call `toLocaleString` with the argument `"en-u-nu-latn"` to ensure that we get the right digit representations regardless of the locale in which the tests are executed.

Reply:
Done.

Reply:
I was unsure about whether to still pass `en-u-nu-latn` to each invocation of toLocaleString like I suggested above. In the end I think it's OK not to; it would clutter up the tests quite a bit and I am sure there are a bunch of other existing tests that would fail if the default locale wasn't `en`.